src/charts/bar.js

```diff
@@ -177,7 +183,11 @@ define(function(require) {
                 drawBars();
                 drawAxis();
 
-                if (enableLabels) {
+                if (enableLabels && locale) {
+                    localeRequest
```

Comment:
Would it work if we set up localeRequest as a resolved promise by default?

Reply:
I'm not sure what you mean, could you explain with some (pseudo) code ?

Reply:
Instead of declaring localeRequest as undefined, you could do:
  localeRequest = Promise.resolve();

Then, you wouldn't need to add the branching here:

   localeRequest.then(() => drawLabels())
         .catch(() => new Error('You need an Internet connection in order to load the selected locale'))

Reply:
I've added the Promise.resolve to the declaration of localeRequest, but i don't quite understand the  branching part.

Reply:
If the value of localeRequest is a resolved promise by default, then you don't need to check for 'enableLables' or 'locale', you can just put the same code for all cases.

Reply:
> If the value of localeRequest is a resolved promise by default, then you don't need to check for 'enableLables' or 'locale', you can just put the same code for all cases.

Got it now, good catch!